functors_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,37 @@
 #include "impl.h"
+#include <array>
+#include <algorithm>
 
+BeerBrand BeerOrganizer::operator()() noexcept {
+    if (m_number==m_max){
+        m_number = m_min+1;
+    }
+    return static_cast<BeerBrand>(m_number++);
+}
+
+BeerOrganizer::BeerOrganizer():m_number(m_min+1) {
+    static_assert(m_max-m_min>1, "You have only None and Max in enum!");
+}
+
+bool isExpensiveBeer(const BeerBrand brand) {
+    constexpr static std::array<BeerBrand, 2> expensiveBeers =
+            {BeerBrand::Corona, BeerBrand::HoeGaarden};
+    return std::find(expensiveBeers.cbegin(), expensiveBeers.cend(), brand)!=expensiveBeers.cend();
+}
+
+Cocktail MixingPolicy::mix(const AlcoholDrink alcoholDrink, const NonAlcoholDrink nonAlcoholDrink) {
+    using Ingredients = std::pair<AlcoholDrink, NonAlcoholDrink>;
+    using Recipe = std::pair<Ingredients , Cocktail>;
+    constexpr static std::array<Recipe , 3> recipes = {
```

Comment:
In such cases, unordered map should be preferred.

But nice try :)

Reply:
according to cppref (https://en.cppreference.com/w/cpp/utility/hash) std::pair has no std::hash

Reply:
So you should provide your own implementation, it is pretty simple